groups/bsl/bslstl/bslstl_ratio.t.cpp

```diff
@@ -0,0 +1,349 @@
+                verbose = argc > 2;
+            veryVerbose = argc > 3;
+        veryVeryVerbose = argc > 4;
+    veryVeryVeryVerbose = argc > 5;
+
+    printf("TEST " __FILE__ " CASE %d\n", test);
+
+    switch (test) { case 0:  // Zero is always the leading case.
+      case 2: {
+        // --------------------------------------------------------------------
+        // TESTING USAGE EXAMPLE
+        //
+        // Concerns:
+        //: 1 The usage example provided in the component header file compiles,
+        //:   links, and runs as shown.
+        //
+        // Plan:
+        //: 1 Incorporate usage example from header into test driver, remove
+        //:   leading comment characters.
+        //
+        // Testing:
+        //   USAGE EXAMPLE
+        // --------------------------------------------------------------------
+
+        if (verbose) printf("\nTesting Usage Example"
+                            "\n=====================\n");
+
+      } break;
+      case 1: {
+        // ----------------------------------------------------------------------
+        // TESTING the ratio operations
+        //
+        // Concerns:
+        //: 1 That the operations are successfully imported into the 'bsl'
+        //:   namespace from the underlying STL implementation.
+        //: 
+        //: 2 That the short versions of the comparison operators, such
+        //:   as 'ratio_less_v' exist and work even if the underlying STL
+        //:   does not provide them.
+        //
+        // Plan:
+        //: 1 Verify that the SI typedefs exist and have the correct 
+        //:   numerator and denominator.
+        //:
+        //: 2 Verify that the comparison operations exist and give 
+        //:   expected results. Check again for the '_v' versions.
+        //:
+        //: 3 Verify that the arithmetic operations exist and give
+        //:   expected results.
+        //
+        // ----------------------------------------------------------------------
+
+        if (verbose) printf("\nTesting ratio operations"
+                            "\n======================================\n");
+
+        if (verbose) printf("\tTest SI unit prefixes.\n");
+        {
+            ASSERT(1 == bsl::deci::num);
+            ASSERT(1 == bsl::centi::num);
+            ASSERT(1 == bsl::milli::num);
+            ASSERT(1 == bsl::micro::num);
+            ASSERT(1 == bsl::nano::num);
+            ASSERT(1 == bsl::pico::num);
+            ASSERT(1 == bsl::femto::num);
+            ASSERT(1 == bsl::atto::num);
+
+            ASSERT(10ULL == bsl::deci::den);
+            ASSERT(100ULL == bsl::centi::den);
+            ASSERT(1000ULL == bsl::milli::den);
+            ASSERT(1000000ULL == bsl::micro::den);
+            ASSERT(1000000000ULL == bsl::nano::den);
+            ASSERT(1000000000000ULL == bsl::pico::den);
+            ASSERT(1000000000000000ULL == bsl::femto::den);
+            ASSERT(1000000000000000000ULL == bsl::atto::den);
```

Comment:
I think we'd align on `ULL ==` here.  There is no rule for this particular case beyond "align to look nice" (so eyes of beholder play into this sort of thing):
```
  ASSERT(                 10ULL == bsl::deci::den);
  ASSERT(                100ULL == bsl::centi::den);
  ...
  ASSERT(   1000000000000000ULL == bsl::femto::den);
  ASSERT(1000000000000000000ULL == bsl::atto::den);
```

Reply:
I like that.